drivers/gnss/gnss_quectel_lx6.c

```diff
@@ -0,0 +1,732 @@
+#ifdef CONFIG_MODEM_BACKEND_UART
+		data->pipe = modem_backend_uart_init(&data->backend.uart,
+						     &config->backend.uart_config);
+#endif
+	}
+}
+
+static void quectel_lx6_init_pmtk_script(const struct device *dev)
+{
+	struct quectel_lx6_data *data = dev->data;
+
+	modem_chat_match_init(&data->pmtk_match);
+	modem_chat_match_set_separators(&data->pmtk_match, ",*");
+
+	modem_chat_script_chat_init(&data->pmtk_script_chat);
+	modem_chat_script_chat_set_response_matches(&data->pmtk_script_chat,
+						    &data->pmtk_match, 1);
+
+	modem_chat_script_init(&data->pmtk_script);
+	modem_chat_script_set_name(&data->pmtk_script, "pmtk");
+	modem_chat_script_set_script_chats(&data->pmtk_script, &data->pmtk_script_chat, 1);
+	modem_chat_script_set_abort_matches(&data->pmtk_script, NULL, 0);
+	modem_chat_script_set_timeout(&data->pmtk_script, 10);
+}
+
+static int quectel_lx6_init(const struct device *dev)
+{
+	const struct quectel_lx6_config *config = dev->config;
+	struct quectel_lx6_data *data = dev->data;
+	int ret;
+
+	k_mutex_init(&data->mutex);
+
+	if (config->vcc.port != NULL) {
+		ret = gpio_pin_configure_dt(&config->vcc, GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't config VCC", dev->name);
+			return ret;
+		}
+	}
+
+#if CONFIG_GNSS_QUECTEL_LX6_RESET_ON_INIT
+	if ((config->reset.port != NULL) && (config->vcc.port != NULL)) {
+		ret = gpio_pin_set_dt(&config->vcc, 1);
+		if (ret < 0) {
+			LOG_ERR("[%s] couldn't config VCC", dev->name);
+			return ret;
+		}
+		ret = gpio_pin_set_dt(&config->reset, 0); /* Inactive is high */
+		if (ret < 0) {
+			LOG_ERR("Failed to inactivate reset pin: %d", ret);
+		}
+		k_msleep(4); /* > 2ms */
+		ret = gpio_pin_set_dt(&config->reset, 1); /* Active is low */
+		if (ret < 0) {
+			LOG_ERR("Failed to activate reset pin: %d", ret);
+		}
+		k_msleep(12); /* Pulldown > 10ms */
+		ret = gpio_pin_set_dt(&config->reset, 0); /* Inactive is high */
+		if (ret < 0) {
+			LOG_ERR("Failed to inactivate reset pin: %d", ret);
+		}
+	} else {
+		LOG_WRN("[%s] couldn't reset", dev->name);
+	}
+#endif
+
+	ret = quectel_lx6_init_nmea0183_match(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lx6_init_pipe(dev);
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = ARRAY_SIZE(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	ret = modem_chat_init(&data->chat, &chat_config);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lx6_init_pmtk_script(dev);
+
+	if (pm_device_is_powered(dev)) {
+		ret = quectel_lx6_resume(dev);
+		if (ret < 0) {
+			return ret;
+		}
+	} else {
+		pm_device_init_off(dev);
+	}
+
+	return pm_device_runtime_enable(dev);
```

Comment:
This has been superceeded by using

```
return pm_device_driver_init(dev, quectel_lx6_pm_action);
```
which includes very similar logic :)

Reply:
Done